### PR TITLE
Support for code generation from material nodes.

### PR DESCRIPTION
**Pull request by niklasharrysson**
This change list adds support for generating shaders directly from `material` type nodes. Previously material nodes were terminals from which we first extracted shader nodes to generate code from. Now we can generate code from these nodes directly and include custom logic for them.

This change is foundational work towards having support for double-sided materials, which will be added in a future change list (issue tracked here https://github.com/AcademySoftwareFoundation/MaterialX/issues/864).

### CODE REVIEW

**Comment by niklasharrysson** [comment]
Test suite renderings for GLSL and OSL after the change.
[tests_new_compressed.pdf](https://github.com/AcademySoftwareFoundation/MaterialX/files/8846257/tests_new_compressed.pdf)

File: source/MaterialXTest/MaterialXRenderOsl/RenderOsl.cpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ class TangentOsl : public mx::ShaderNodeImpl
         BEGIN_SHADER_STAGE(stage, mx::Stage::PIXEL)
             shadergen.emitLineBegin(stage);
             shadergen.emitOutput(node.getOutput(), true, false, context, stage);
-            shadergen.emitString(" = normalize(vector(N.z, 0, -N.x))", stage);
+            shadergen.emitString(" = normalize(vector(N[2], 0, -N[0]))", stage);
```
Comment (niklasharrysson) [inline source/MaterialXTest/MaterialXRenderOsl/RenderOsl.cpp:43]: Fix to support OSL back to 1.9.10.

File: source/MaterialXView/Viewer.cpp [hunk 2]
```diff
@@ -1223,19 +1223,7 @@ void Viewer::loadDocument(const mx::FilePath& filename, mx::DocumentPtr librarie
         {
             mx::TypedElementPtr renderableElem = elem;
             mx::NodePtr node = elem->asA<mx::Node>();
-            if (node && node->getType() == mx::MATERIAL_TYPE_STRING)
```
Comment (niklasharrysson) [inline source/MaterialXView/Viewer.cpp:1226]: No need for special treatment of material nodes now.

**Comment by jstone-lucasfilm** [comment]
This looks like a great step forward, thanks @niklasharrysson!  I should note that it appears to successfully address report #620, which was written up last year by @numega.  I'll plan to take a closer look and merge the change if no issues are found.

**Comment by jstone-lucasfilm** [comment]
Updated reference renders, with transparency on BrassWireMesh material:
[MaterialXReferenceRenders_06_07_2022.pdf](https://github.com/AcademySoftwareFoundation/MaterialX/files/8857070/MaterialXReferenceRenders_06_07_2022.pdf)

**Review by jstone-lucasfilm — APPROVED** [review]
Looks good to me, thanks @niklasharrysson!

**Comment by numega** [comment]
Awesome! Thanks for taking care of this @niklasharrysson

**Pull request merged**